tests/test_constructors.py

```diff
@@ -0,0 +1,156 @@
+@pytest.mark.parametrize('use_stokes', [True, False])
+@pytest.mark.parametrize('use_wind', [True, False])
+@pytest.mark.parametrize('use_mixing', [True, False])
+def test_create_fieldset(use_3D, use_biofouling, use_stokes, use_wind, use_mixing):
+    settings_file = 'tests/test_data/test_settings.json'
+    settings = pp.utils.load_settings(settings_file)
+    settings = pp.utils.download_plasticparcels_dataset('NEMO0083', settings, 'input_data')
+
+    settings['simulation'] = make_standard_simulation_settings()
+
+    settings['use_3D'] = use_3D
+    settings['use_biofouling'] = use_biofouling
+    settings['use_stokes'] = use_stokes
+    settings['use_wind'] = use_wind
+    settings['use_mixing'] = use_mixing
+
+    fieldset = pp.constructors.create_fieldset(settings)
+
+    assert isinstance(fieldset, parcels.FieldSet)
+
+
+# Test three different initialisation maps
+@pytest.mark.parametrize('initialisation_map', ['coastal', 'fisheries', 'rivers'])
+def test_create_particleset_from_map(initialisation_map):
+    settings_file = 'tests/test_data/test_settings.json'
+    settings = pp.utils.load_settings(settings_file)
+    settings = pp.utils.download_plasticparcels_dataset('NEMO0083', settings, 'input_data')
+
+    settings['simulation'] = make_standard_simulation_settings()
+
+    settings['plastictype'] = make_standard_plastictype_settings()
+
+    settings['release'] = {'initialisation_type': initialisation_map,
+                           'country': 'Italy',
+                           }
+
+    fieldset = make_simple_fieldset()
+    pset = pp.constructors.create_particleset_from_map(fieldset, settings)
+
+    assert isinstance(pset, parcels.ParticleSet)
+
+
+# Test the two global concentration release map types
+@pytest.mark.parametrize('concentration_type', ['Beach', 'Ocean'])
+def test_create_particleset_from_map_concentrations(concentration_type):
+    settings_file = 'tests/test_data/test_settings.json'
+    settings = pp.utils.load_settings(settings_file)
+    settings = pp.utils.download_plasticparcels_dataset('NEMO0083', settings, 'input_data')
+
+    settings['simulation'] = make_standard_simulation_settings()
+
+    settings['plastictype'] = make_standard_plastictype_settings()
+
+    settings['release'] = {'initialisation_type': 'global_concentrations',
+                           'concentration_type': concentration_type,
+                           }
+
+    fieldset = make_simple_fieldset()
+    pset = pp.constructors.create_particleset_from_map(fieldset, settings)
+
+    assert isinstance(pset, parcels.ParticleSet)
+
+
+# Test three different initialisation maps
+@pytest.mark.parametrize('use_3D', [True, False])
+@pytest.mark.parametrize('use_biofouling', [True, False])
+@pytest.mark.parametrize('use_stokes', [True, False])
+@pytest.mark.parametrize('use_wind', [True, False])
+@pytest.mark.parametrize('use_mixing', [True, False])
+def test_create_kernel(use_3D, use_biofouling, use_stokes, use_wind, use_mixing):
```

Comment:
I've thought of this as an extension to the test above. If the one above fails, then this will clearly fail too. But the test above may correctly create a fieldet, and this test is checking if the relevant kernels are then added to the list. One test for fieldset creation, one test for kernel list creation, was the idea.